src/highdicom/seg/sop.py

```diff
@@ -61,6 +75,723 @@
+
+        Returns
+        -------
+        bool
+            True if the ReferencedSOPInstanceUID (in combination with the
+            segment number) uniquely identifies frames of the segmentation
+            image.
+
+        """
+        cur = self._db_con.cursor()
+        n_unique_combos = cur.execute(
+            'SELECT COUNT(*) FROM '
+            '(SELECT 1 FROM FrameLUT GROUP BY ReferencedSOPInstanceUID, '
+            'SegmentNumber)'
+        ).fetchone()[0]
+        return n_unique_combos == self._number_of_frames
+
+    def are_referenced_frames_unique(self) -> bool:
+        """Check if ReferencedFrameNumber uniquely identify frames.
+
+        Returns
+        -------
+        bool
+            True if the ReferencedFrameNumber (in combination with the
+            segment number) uniquely identifies frames of the segmentation
+            image.
+
+        """
+        cur = self._db_con.cursor()
+        n_unique_combos = cur.execute(
+            'SELECT COUNT(*) FROM '
+            '(SELECT 1 FROM FrameLUT GROUP BY ReferencedFrameNumber, '
+            'SegmentNumber)'
+        ).fetchone()[0]
+        return n_unique_combos == self._number_of_frames
+
+    def get_unique_sopuids(self) -> Set[str]:
+        """Get set of unique Referenced SOP Instance UIDs.
+
+        Returns
+        -------
+        Set[str]
+            Set of unique Referenced SOP Instance UIDs.
+
+        """
+        cur = self._db_con.cursor()
+        return {
+            r[0] for r in
+            cur.execute(
+                'SELECT DISTINCT(SOPInstanceUID) from InstanceUIDs'
+            )
+        }
+
+    def get_max_frame_number(self) -> int:
+        """Get highest frame number of any referenced frame.
+
+        Returns
+        -------
+        int
+            Highest frame number referenced in the segmentation image.
+
+        """
+        cur = self._db_con.cursor()
+        return cur.execute(
+            'SELECT MAX(ReferencedFrameNumber) FROM FrameLUT'
+        ).fetchone()[0]
+
+    def get_unique_dim_index_values(
+        self,
+        dimension_index_pointers: Sequence[int],
+    ) -> Set[Tuple[int, ...]]:
+        """Get set of unique dimension index value combinations.
+
+        Parameters
+        ----------
+        dimension_index_pointers: Sequence[int]
+            List of dimension index pointers for which to find unique
+            combinations of values.
+
+        Returns
+        -------
+        Set[Tuple[int, ...]]
+            Set of unique dimension index values for the input dimension index
+            pointers.
+
+        """
+        cols = [self._dim_ind_col_names[p] for p in dimension_index_pointers]
+        cols_str = ', '.join(cols)
+        cur = self._db_con.cursor()
+        return {
+            r for r in
+            cur.execute(
+                f'SELECT DISTINCT {cols_str} FROM FrameLUT'
+            )
+        }
+
+    @contextmanager
+    def temp_table(
```

Comment:
Should this method and the following be private methods (i.e., starting with `_`)?

I would also consider renaming them `_generate_temp_table`. I generally prefer function names to start with a verb.